Behavior difference: Callback-first promise resolve() gives inconsistent error

## Steps to reproduce

1. Start a plain new project
    - "@typescript/native-preview": "7.0.0-dev.20260428.1",
    - "typescript": "6.0.3"
2. Run `npx tsgo --init` to get the default tsconfig. Turn on these flags as well.
    - "allowJs": true,
    - "checkJs": true,
    - "stableTypeOrdering": true,
3. Add `src/main.js`. Run `npx tsc --noEmit` & `npx tsgo --noEmit` to see the difference.

```js
// src/main.js
new Promise((resolve) => resolve());
```

## Behavior with `typescript@6.0`

```
src/main.js:2:26 - error TS2810: Expected 1 argument, but got 0. 'new Promise()' needs a JSDoc hint to produce a 'resolve' that can be called without arguments.

2 new Promise((resolve) => resolve());
                           ~~~~~~~


Found 1 error in src/main.js:2
```

## Behavior with `tsgo`

```
src/main.js:2:26 - error TS2794: Expected 1 arguments, but got 0. Did you forget to include 'void' in your type argument to 'Promise'?

2 new Promise((resolve) => resolve());
                           ~~~~~~~

  node_modules/.../lib.es2015.promise.d.ts:29:34 - An argument for 'value' was not provided.
    29     new <T>(executor: (resolve: (value: T | PromiseLike<T>) => void, reject: (reason?: any) => void) => void): Promise<T>;
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in src/main.js:2
```

---

I think TS 6 makes more sense in this case, since `type argument` is not really a thing inside a javascript file.

--- a/internal/checker/checker.go
+++ b/internal/checker/checker.go
@@ -9559,6 +9559,10 @@ func (c *Checker) getArgumentArityError(node *ast.Node, signatures []*Signature,
 		parameterRange = strconv.Itoa(minCount)
 	}
 	isVoidPromiseError := !hasRestParameter && parameterRange == "1" && len(args) == 0 && c.isPromiseResolveArityError(node)
+	errorNode := getErrorNodeForCallNode(node)
+	if isVoidPromiseError && ast.IsInJSFile(node) {
+		return NewDiagnosticForNode(errorNode, diagnostics.Expected_1_argument_but_got_0_new_Promise_needs_a_JSDoc_hint_to_produce_a_resolve_that_can_be_called_without_arguments)
+	}
 	var message *diagnostics.Message
 	switch {
 	case ast.IsDecorator(node):
@@ -9574,7 +9578,6 @@ func (c *Checker) getArgumentArityError(node *ast.Node, signatures []*Signature,
 	default:
 		message = diagnostics.Expected_0_arguments_but_got_1
 	}
-	errorNode := getErrorNodeForCallNode(node)
 	switch {
 	case minCount < len(args) && len(args) < maxCount:
 		// between min and max, but with no matching overload

--- a/testdata/baselines/reference/compiler/jsPromiseNeedsJSDocHint.errors.txt
+++ b/testdata/baselines/reference/compiler/jsPromiseNeedsJSDocHint.errors.txt
@@ -1,0 +1,8 @@
+main.js(1,26): error TS2810: Expected 1 argument, but got 0. 'new Promise()' needs a JSDoc hint to produce a 'resolve' that can be called without arguments.
+
+
+==== main.js (1 errors) ====
+    new Promise((resolve) => resolve());
+                             ~~~~~~~
+!!! error TS2810: Expected 1 argument, but got 0. 'new Promise()' needs a JSDoc hint to produce a 'resolve' that can be called without arguments.
+    

--- a/testdata/baselines/reference/compiler/jsPromiseNeedsJSDocHint.symbols
+++ b/testdata/baselines/reference/compiler/jsPromiseNeedsJSDocHint.symbols
@@ -1,0 +1,8 @@
+//// [tests/cases/compiler/jsPromiseNeedsJSDocHint.ts] ////
+
+=== main.js ===
+new Promise((resolve) => resolve());
+>Promise : Symbol(Promise, Decl(lib.es5.d.ts, --, --), Decl(lib.es2015.iterable.d.ts, --, --), Decl(lib.es2015.promise.d.ts, --, --), Decl(lib.es2015.symbol.wellknown.d.ts, --, --), Decl(lib.es2018.promise.d.ts, --, --))
+>resolve : Symbol(resolve, Decl(main.js, 0, 13))
+>resolve : Symbol(resolve, Decl(main.js, 0, 13))
+

--- a/testdata/baselines/reference/compiler/jsPromiseNeedsJSDocHint.types
+++ b/testdata/baselines/reference/compiler/jsPromiseNeedsJSDocHint.types
@@ -1,0 +1,11 @@
+//// [tests/cases/compiler/jsPromiseNeedsJSDocHint.ts] ////
+
+=== main.js ===
+new Promise((resolve) => resolve());
+>new Promise((resolve) => resolve()) : Promise<any>
+>Promise : PromiseConstructor
+>(resolve) => resolve() : (resolve: (value: any) => void) => void
+>resolve : (value: any) => void
+>resolve() : void
+>resolve : (value: any) => void
+

--- a/testdata/tests/cases/compiler/jsPromiseNeedsJSDocHint.ts
+++ b/testdata/tests/cases/compiler/jsPromiseNeedsJSDocHint.ts
@@ -1,0 +1,7 @@
+// @checkJs: true
+// @noEmit: true
+
+// https://github.com/microsoft/typescript-go/issues/3641
+
+// @filename: main.js
+new Promise((resolve) => resolve());

